### PR TITLE
Badging Engine Part #2 - More Notifications!

### PR DESCRIPTION
**Pull request by cornehoskam**
As per #14, added even more notifications, and their Failed counterparts! The various properties of the failed notification may need to be updated in the future depending on the usecases they're wished to be used for, but that's most likely a concern for the future! 🙂

### CODE REVIEW

File: Quiz.Site/Controllers/Surface/ProfileSurfaceController.cs [hunk 1]
```diff
@@ -149,23 +150,24 @@ public async Task<IActionResult> EditProfile(EditProfileViewModel model)
                         _logger.LogError($" - {item.Key}: {item.Value}");
                     }
                 }
+                await _eventAggregator.PublishAsync(new ProfileUpdatingFailedNotification("Edit Profile Model State Invalid"));
                 return RedirectToCurrentUmbracoPage();
             }
-
-            var user = await _memberManager.GetCurrentMemberAsync();
```
Comment (cornehoskam) [inline Quiz.Site/Controllers/Surface/ProfileSurfaceController.cs:155]: This variable wasn't used, so I took the liberty to remove it!

File: Quiz.Site/Controllers/Surface/ProfileSurfaceController.cs [hunk 2]
```diff
@@ -191,11 +193,7 @@ public async Task<IActionResult> EditProfile(EditProfileViewModel model)
                     TempData["ShowToast"] = true;
                 }
             }
-
-            var profilePage = CurrentPage.AncestorOrSelf<HomePage>().FirstChildOfType(ProfilePage.ModelTypeAlias);
```
Comment (cornehoskam) [inline Quiz.Site/Controllers/Surface/ProfileSurfaceController.cs:195]: Same with this one, as we're currently on the profilePage, this variable wasn't used.

Comment (prjseal) [reply Quiz.Site/Controllers/Surface/ProfileSurfaceController.cs:195]: Brill thanks

**Review by prjseal — APPROVED** [review]

**Comment by prjseal** [comment]
Looking good, laying good foundations here

**Pull request merged**